fdbserver/workloads/TagThrottleApi.actor.cpp

```diff
@@ -137,7 +138,8 @@ struct TagThrottleApiWorkload : TestWorkload {
 			}
 		}
 
-		bool removed = wait(ThrottleApi::unthrottleTags(cx, tagSet, throttleType, priority));
+		state Reference<Database> db = makeReference<Database>(cx);
+		bool removed = wait(ThrottleApi::unthrottleTags(db, tagSet, throttleType, priority));
```

Comment:
This is an awkward usage of the `Reference` API, and sadly it's almost but not actually unnecessary. `Database` is just a wrapper around a `Reference<DatabaseContext>`, and using it would be cleaner if it were possible to implement `createTransaction` on it cleanly. To be honest, I'm not even actually sure what purpose the `Database` wrapper serves.

Obviously we could solve this by eliminating the wrapper, though I don't know if that's the right move or not.

Reply:
I somehow started with using `Reference<DatabaseContext>` but changed to using `Database`.
It's honestly just a wrapper of a pointer to DatabaseContext,
the only reason I changed to it is it's a cleaner a small interface easier to understand by taking a quick look.
And _Database_ looks comparable to `IDatabase` while using the template function.

Reply:
The problem with using `DatabaseContext` is that you can't straightforwardly implement `createTransaction` on it. The `ReadYourWritesTransaction` constructor requires a `Database` object to create it, even though it is of course just using data from the underlying `DatabaseContext`. 

You could make a version of the constructor that takes a `DatabaseContext` and wraps it with a `Database`, which is also a little awkward but potentially more straightforward than making `Database` reference counted. I'd say if we aren't going to deal with getting rid of the wrapper class now, this is probably the cleanest version.

Reply:
Yeah, it makes sense.
I agree it seems to be the cleanest way if we keep _Database_, though I still feel awkward this way.
And thanks for catching this, 
making _Database_ reference-counted looks weird to me, 
I was hoping to get advice on it and forgot to mention it in the description.

Reply:
Changed to use _DatabaseContext_